lightning/backend/local.go

```diff
@@ -1112,6 +1218,10 @@ func (local *local) ImportEngine(ctx context.Context, engineUUID uuid.UUID) erro
 	}
 
 	lf := engineFile.(*LocalFile)
+
+	lf.lock(importMutexStateImport)
+	defer lf.unlock()
+
```

Comment:
we might reload engine file from `local.engines` here? or we might always lock the **UUID** (maybe a `map[UUID]mutex` like `local.engines`, to ge rid of getting the file -> checking the lock -> getting file again) before accessing the `LocalFile`?

And not sure `ResetEngine` should hold that UUID lock all through the function

Reply:
disk quota routine is concurrent with normal open-write-close-import routine, and engines are flush/close/import/reopen in two rontine but at same UUID, that may make things easier to understood 😵 

and maybe UUID level lock instead of `rc.diskQuotaLock` could avoid blocking all engines' writing

Reply:
But the requirement is to lock all engines to avoid any further writing to disk since the disk is near full..

Reply:
I was recalling https://github.com/pingcap/tidb-lightning/pull/493#discussion_r539021798

if the "soft limit" works well, we could only block the `largeEngines` returned in `CheckDiskQuota`